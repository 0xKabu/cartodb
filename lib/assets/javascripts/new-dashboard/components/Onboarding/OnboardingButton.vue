<template>
  <button
    class="button"
    :class="[isFirstTimeViewingDashboard ? 'button--cta' : 'button--ghost']"
    @click="openOnboarding()">
    {{ $t('Wizards.Distributor.cta') }}
  </button>
</template>

<script>
export default {
  name: 'OnboardingButton',
  props: {
    isFirstTimeViewingDashboard: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    openOnboarding () {
      this.$router.push({ name: 'onboarding' });
    }
  }
};
</script>
