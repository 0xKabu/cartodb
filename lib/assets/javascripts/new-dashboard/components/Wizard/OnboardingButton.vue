<template>
  <button
    class="button"
    :class="[isFirstTimeViewingDashboard ? 'button--cta' : 'button--ghost']"
    @click="openOnboarding()">
    {{ $t('Wizards.Distributor.cta') }}
  </button>
</template>

<script>
export default {
  name: 'OnboardingButton',
  props: {
    isFirstTimeViewingDashboard: Boolean
  },
  methods: {
    openOnboarding () {
      this.$router.push({ name: 'onboarding' });
    }
  }
};
</script>
<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.button {
  &.button--ghost {
    margin-right: 36px;
    padding: 0;
  }
}
</style>
