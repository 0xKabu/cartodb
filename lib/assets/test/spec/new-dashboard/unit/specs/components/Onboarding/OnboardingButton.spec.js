import { shallowMount } from '@vue/test-utils';
import OnboardingButton from 'new-dashboard/components/Onboarding/OnboardingButton';

const $t = key => key;

describe('OnboardingButton.vue', () => {
  it('should render properly', () => {
    const onboardingButton = shallowMount(OnboardingButton, {
      mocks: { $t }
    });

    expect(onboardingButton).toMatchSnapshot();
  });

  it('should set button--cta if isFirstTimeViewingDashboard is true', () => {
    const onboardingButton = shallowMount(OnboardingButton, {
      propsData: {
        isFirstTimeViewingDashboard: true
      },
      mocks: { $t }
    });

    expect(onboardingButton).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should call router.push when opening onboarding', () => {
      const $router = {
        push: jest.fn()
      };

      const onboardingButton = shallowMount(OnboardingButton, {
        mocks: { $t, $router }
      });

      onboardingButton.vm.openOnboarding();

      expect($router.push).toHaveBeenCalledWith({ name: 'onboarding' });
    });
  });
});
