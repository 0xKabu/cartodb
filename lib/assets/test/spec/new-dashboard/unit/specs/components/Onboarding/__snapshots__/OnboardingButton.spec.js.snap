// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingButton.vue should render properly 1`] = `
<button class="button button--ghost">
  Wizards.Distributor.cta
</button>
`;

exports[`OnboardingButton.vue should set button--cta if isFirstTimeViewingDashboard is true 1`] = `
<button class="button button--cta">
  Wizards.Distributor.cta
</button>
`;
